### PR TITLE
Create/Edit Request Forms

### PR DESCRIPTION
**Pull request by meganniu**
**UI IS NOT DONE** Type and quantity fields do not match designs. I am waiting for those components to be finished.

Blocked by:
+ SearchableTagDropdown
+ InfiniteDropdown

Testing URLS:
+ create request form:
  + http://localhost:3000/create-request-form
  + http://localhost:3000/create-request-form
+ edit request form:
  + http://localhost:3000/edit-request-form

Changes:
+ server:
  + createClient behaviour:
    + If client's fullName is given, then find id of client with the fullName. If no client exists with the given fullName, create the client
    + add the found/created client to the request before creating the request
  + updateClient bhaviour:
    + If client's fullName is given, then find id of client with the fullName. If no client exists with the given fullName, create the client
    + add the found/created client to the request before updating the request
+ client:
  + added Request forms

### CODE REVIEW

File: client/src/components/atoms/SearchableDropdown.tsx [hunk 1]
```diff
@@ -1,24 +1,29 @@
-import React, { FunctionComponent, useState } from "react";
+import React, { FunctionComponent, useEffect, useState } from "react";
```
Comment (meganniu) [inline client/src/components/atoms/SearchableDropdown.tsx:1]: ignore this file for now
i had to make tons of modifcations, but most edits will be reverted once the SearchableTagInput component is done

File: server/models/clientModel.ts [hunk 2]
```diff
@@ -11,8 +11,7 @@ type ClientDocument = ClientInterface & Document;
 
 const ClientSchema = new Schema({
   clientId: {
-      type: String,
-      required: true
+    type: String,
```
Comment (kevin-wangg) [inline server/models/clientModel.ts:14]: How come this field is no longer required?

**Review by kevin-wangg — APPROVED** [review]
LGTM! I just left a comment regarding a question I had.

File: server/models/clientModel.ts (hunk 2 above)
Comment (meganniu) [reply server/models/clientModel.ts:14]: this was supposed to be the ekyros id of the client. 
I don't want to put any garbage/placeholder default value, so Im just leaving it out of all client objects until we know what data we're storing on clients

**Review by chamod-gamage — APPROVED** [review]
LGTM so far!!!

**Comment by bonnie-chin** [comment]
Looks good! Any bugs / comments we noticed are all inside of the taskboard
- https://www.notion.so/uwblueprintexecs/Bug-Create-Request-Form-groups-aren-t-in-alphabetical-order-a774c7c8f33c4ba1a11a06c4f66c4e6f (Bonnie)
- https://www.notion.so/uwblueprintexecs/Bug-Once-a-group-or-type-is-selected-in-the-create-request-form-no-others-can-be-selected-cda8e840f8d747b2aad438f5fbdf6183 (Ainley)

**Pull request merged**